pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -357,27 +372,49 @@ exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
 			*caBundle))
 	}
 
+	clusterDNSIP := p.clusterDNSIP
```

Comment:
I think we'd also need to constrain the instance types that we can use when doing ipv6. Only nitro instance types support IPv6 Prefix Delegation.

Reply:
Yes we need to launch Nitro instances to use IPv6. However I think it's very useful if Karpenter has a function to be able to filter non-Nitro instances by using provisioner's constraints.
Currently we need to specify instance types which are on Nitro by provisioner settings.